bsmetadata/evaluation.py

```diff
@@ -197,12 +328,70 @@ def get_ppl(batch: Dict[str, torch.Tensor]) -> torch.Tensor:
                 if not args.no_cuda:
                     normal_batch = {k: v.cuda() for k, v in normal_batch.items()}
                     metadata_batch = {k: v.cuda() for k, v in metadata_batch.items()}
+                if n_examples == 1:
+                    ex = format_by_one_mask(normal_batch["input_ids"][0], normal_batch["attention_mask"][0], tokenizer)
+                    rich.print(f"Normal example:")
+                    rich.print(ex)
+
+                    ex = format_by_one_mask(
+                        metadata_batch["input_ids"][0], metadata_batch["metadata_mask"][0], tokenizer
+                    )
+                    rich.print(f"Metadata example:")
+                    rich.print(ex)
+                    rich.print(tokenizer.decode(metadata_batch["input_ids"][0]))
 
                 # Calculate ppl
-                normal_ppl = get_ppl(normal_batch)
-                total_normal_ppl += float(normal_ppl) * normal_example_len
-                metadata_ppl = get_ppl(metadata_batch)
-                total_metadata_ppl += float(metadata_ppl) * metadata_example_len
+                normal_ppl = get_ppl(normal_batch, save_data=args.save_data, idx=idx)
+                # total_normal_ppl += float(normal_ppl) * normal_example_len
+                metadata_ppl = get_ppl(metadata_batch, save_data=args.save_data, idx=idx)
+                # total_metadata_ppl += float(metadata_ppl) * metadata_example_len
```

Comment:
Why did you comment it out? Same above

Reply:
Hmmm, I don't remember. I think I just wanted to see the ppl of the first example, but I don't know why I commented it out.